internal/gcsx/read_manager/read_manager.go

```diff
@@ -65,6 +72,29 @@ func NewReadManager(object *gcs.MinObject, bucket gcs.Bucket, config *ReadManage
 		readers = append(readers, fileCacheReader) // File cache reader is prioritized.
 	}
 
+	// If buffered read is enabled, initialize the buffered reader and add it to the readers.
+	readConfig := config.Config.Read
+	if config.Config.Read.EnableBufferedRead {
+		bufferedReadConfig := &bufferedread.BufferedReadConfig{
+			MaxPrefetchBlockCnt:     readConfig.MaxBlocksPerHandle,
+			PrefetchBlockSizeBytes:  readConfig.BlockSizeMb * 1024 * 1024,
+			InitialPrefetchBlockCnt: readConfig.StartBlocksPerHandle,
+		}
+		bufferedReader, err := bufferedread.NewBufferedReader(
+			object,
+			bucket,
+			bufferedReadConfig,
+			config.GlobalMaxBlocksSem,
+			config.WorkerPool,
+			config.MetricHandle,
+		)
+		if err != nil {
+			logger.Warnf("Failed to create bufferedReader: %v. Falling back to another reader.", err)
```

Comment:
nit: Please correct the log, here we are not enabling buffered-reader. Not falling back.